lib/worker/base.js

```diff
@@ -19,6 +19,35 @@ import {flags, refs, sharedWorkerTeardowns} from './state.cjs';
 import {isRunningInThread, isRunningInChildProcess} from './utils.cjs';
 
 const currentlyUnhandled = setUpCurrentlyUnhandled();
+let runner = Object.create(null);
+
+// Override process.exit with an undetectable replacement
+// to report when it is called from a test (which it should never be).
+const {apply} = Reflect;
+const realExit = process.exit;
+
+async function exit(code, forceSync = false) {
+	dependencyTracking.flush();
+	if (!forceSync) {
+		await channel.flush();
+	}
+
+	apply(realExit, process, [code]);
+}
+
+process.exit = new Proxy(realExit, {
+	apply(fn, receiver, args) {
+		channel.send({type: 'internal-error', err: serializeError('Forced exit error', false, new Error('Unexpected process.exit()'), runner.file)});
```

Comment:
Could leave `runner` undefined and use `runner?.file`, that should work in the supported Node.js versions I believe.

Reply:
Done.